app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -107,6 +107,49 @@ XrResult OpenXRInputSource::Initialize()
 
         RETURN_IF_XR_FAILED(OpenXRExtensions::sXrCreateHandTrackerEXT(mSession, &handTrackerInfo,
                                                                       &mHandTracker));
+
+        if (OpenXRExtensions::IsExtensionSupported(XR_FB_HAND_TRACKING_MESH_EXTENSION_NAME) &&
+                OpenXRExtensions::sXrGetHandMeshFB != XR_NULL_HANDLE) {
+            XrHandTrackingMeshFB mesh = { XR_TYPE_HAND_TRACKING_MESH_FB };
+            // Figure out sizes first
+            mesh.jointCapacityInput = 0;
+            mesh.vertexCapacityInput = 0;
+            mesh.indexCapacityInput = 0;
+            CHECK_XRCMD(OpenXRExtensions::sXrGetHandMeshFB(mHandTracker, &mesh));
+            mesh.jointCapacityInput = mesh.jointCountOutput;
+            mesh.vertexCapacityInput = mesh.vertexCountOutput;
+            mesh.indexCapacityInput = mesh.indexCountOutput;
+
+            // Skeleton
+            mHandMesh.jointCount = mesh.jointCountOutput;
+            mHandMesh.jointPoses.resize(mesh.jointCountOutput);
+            mHandMesh.jointParents.resize(mesh.jointCountOutput);
+            mHandMesh.jointRadii.resize(mesh.jointCountOutput);
+            mesh.jointBindPoses = mHandMesh.jointPoses.data();
+            mesh.jointParents = mHandMesh.jointParents.data();
+            mesh.jointRadii = mHandMesh.jointRadii.data();
+            // Vertex
+            mHandMesh.vertexCount = mesh.vertexCountOutput;
+            mHandMesh.vertexPositions.resize(mesh.vertexCountOutput);
+            mHandMesh.vertexNormals.resize(mesh.vertexCountOutput);
+            mHandMesh.vertexUVs.resize(mesh.vertexCountOutput);
+            mHandMesh.vertexBlendIndices.resize(mesh.vertexCountOutput);
+            mHandMesh.vertexBlendWeights.resize(mesh.vertexCountOutput);
+            mesh.vertexPositions = mHandMesh.vertexPositions.data();
+            mesh.vertexNormals = mHandMesh.vertexNormals.data();
+            mesh.vertexUVs = mHandMesh.vertexUVs.data();
+            mesh.vertexBlendIndices = mHandMesh.vertexBlendIndices.data();
+            mesh.vertexBlendWeights = mHandMesh.vertexBlendWeights.data();
+            // Index
+            mHandMesh.indexCount = mesh.indexCountOutput;
+            mHandMesh.indices.resize(mesh.indexCountOutput);
+            mesh.indices = mHandMesh.indices.data();
```

Comment:
I was a bit scared by all those resize calls. I checked the specs and they mention that "the data enumerated by this call is constant during the lifetime of an XrInstance" so we shouldn't have to resize them apart from the initialization, shouldn't we?

Reply:
Yes, this is only used one time at initialization, and they don't change. Later on every frame, we just apply a transform to each vertex based on the joints location data.

BTW, I copied this code from the example in the Oculus Mobile SDK, (I case it aims review :)).

Reply:
Oh my bad, it's done in the `Initialize` method, I somehow thought it was in the `Update` one, that's why I got scared. Please forgive my stupidity :)